### PR TITLE
Feature / Pickup campaign code in embark

### PR DESCRIPTION
**Pull request by amandaflod**
## What?

- Create AddCampaignCode mutation to save campaign in quote cart

## Why?

To be able to pick up the campaign code on the offerPage 😊

**Ticket(s): [GRW-602]**

 https://web-onboardi-grw-602-fe-vpq1sx.herokuapp.com/

[GRW-602]: https://hedvig.atlassian.net/browse/GRW-602?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/client/data/graphql.tsx [hunk 1]
```diff
@@ -9978,7 +9979,14 @@ export type SwedishApartmentQuoteDetails = {
...
-  type: ApartmentType
+  type: SwedishApartmentType
+}
+
+export enum SwedishApartmentType {
+  StudentRent = 'STUDENT_RENT',
+  Rent = 'RENT',
+  StudentBrf = 'STUDENT_BRF',
+  Brf = 'BRF',
 }
```
Comment (robinandeer) [inline src/client/data/graphql.tsx:9990]: @amandaflod guess we didn't remove this in the end 😅

Comment (amandaflod) [reply src/client/data/graphql.tsx:9990]: oh no! will fix :D

File: src/client/pages/Embark/index.tsx [hunk 2]
```diff
@@ -237,11 +241,24 @@ const useCreateQuoteCartId = ({ skip = false }) => {
...
+
+  const createQuoteCart = useCallback(async () => {
+    const result = await createOnboardingQuoteCart({
       variables: { market: apiMarket, locale: isoLocale },
     })
-  }, [createOnboardingQuoteCart, apiMarket, isoLocale])
+    const quoteCartId = result.data?.onboardingQuoteCart_create.id
+    const savedCampaignCode = CampaignCode.get()
+
+    if (quoteCartId && savedCampaignCode !== null) {
```
Comment (guilhermespopolin) [inline src/client/pages/Embark/index.tsx:253]: Remember to remove the `savedCampaignCode` in case `addCampaignCode` mutation fails

Comment (amandaflod) [reply src/client/pages/Embark/index.tsx:253]: Great!

**Review by guilhermespopolin — APPROVED** [review]
🚀

**Pull request merged**